kernel/aster-nix/src/fs/inode_handle/mod.rs

```diff
@@ -184,6 +185,36 @@ impl InodeHandle_ {
         self.dentry.inode().poll(mask, poller)
     }
 
+    fn fallocate(&self, flags: FallocateFlags, offset: usize, len: usize) -> Result<()> {
+        let access_mode = self.access_mode();
+        if !access_mode.is_writable() {
+            return_errno_with_message!(Errno::EBADF, "try to fallocate a read-only file");
+        }
```

Comment:
Please move this `access_mode` checking code to `dyn_cap.rs`.

Reply:
Make sense and done